app/components/works/agreement_component.html.erb

```diff
@@ -2,7 +2,7 @@
   <header>Terms of Deposit *</header>
   <div class="row">
     <div class="form-check col-auto">
-      <% if agree_to_terms? %>
+      <% if agree_to_terms? && form.object.model.fetch(:work).depositor.last_work_terms_agreement %>
```

Comment:
this will avoid the nil error in the app.  It's possible there's a better way, but this gets the job done.  (It's not just good, it's good enough.)

Reply:
ugh, despite my better judgement i want to spend more time trying to understand why it doesn't work the original way ... though agreed this will at least avoid the nasty exception (maybe drop a comment in there pointing to this ticket so if we don't fix it now, we will know why it was done later)

Reply:
psst - peter - let it go.  :-)